tests/js/common/shell/shell-cluster-collection.js

```diff
@@ -653,6 +653,66 @@ function ClusterCollectionSuite () {
       }
     },
 
+    testCreateFailureWhenRemovingIsBuilding : function () {
+      if (!isServer) {
+        console.info('Skipping client test');
+        // TODO make client tests work
+        return;
+      }
+      let setFailAt;
+      let removeFailAt;
+      if (isServer) {
+        if (internal.debugCanUseFailAt()) {
+          setFailAt = internal.debugSetFailAt;
+          removeFailAt = internal.debugRemoveFailAt;
+        }
+      } else {
+        const arango = internal.arango;
+        const coordinatorEndpoint = arango.getEndpoint();
+        if (debugCanUseFailAt(coordinatorEndpoint)) {
+          setFailAt = failurePoint => debugSetFailAt(coordinatorEndpoint, failurePoint);
+          removeFailAt = failurePoint => debugRemoveFailAt(coordinatorEndpoint, failurePoint);
+        }
+      }
+      if (!setFailAt) {
+        console.info('Failure tests disabled, skipping...');
+        return;
+      }
+
+      const failurePoint = 'ClusterInfo::createCollectionsCoordinatorRemoveIsBuilding';
+      try {
+        setFailAt(failurePoint);
+        const colName = "UnitTestClusterShouldNotBeCreated1";
+        let threw = false;
+        try {
+          db._create(colName);
+        } catch (e) {
+          threw = true;
+          if (isServer) {
+            assertTrue(e instanceof ArangoError);
+            assertEqual(503, e.errorNum);
+          } else {
+            const expected = {
+              'error': true,
+              'errorNum': 22,
+              'code': 503,
+              'errorMessage': 'intentional debug error',
```

Comment:
I assume this isn't yet correct for the client tests?

Reply:
Checking.